src/gate_detect/src/gate_detect_node.cpp

```diff
@@ -0,0 +1,17 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: February 27th, 2018
+ * Description: Initialize ros node
+ */
+
+#include "GateDetectionNode.h"
+
+using namespace cv;
+
+int main(int argc, char** argv) {
+    ros::init(argc, argv, "gate_detect_node");
+
+    GateDetectionNode gateDetect = GateDetectionNode(argc, argv);
+
+    return 0;
```

Comment:
I think you probably need call `ros::spin()` before you return, otherwise this will just immediately exit?